### PR TITLE
Add path-vars extension

### PR DESCRIPTION
**Pull request by dfrommi**
Extension to replace variable placeholders in query-path like hx-get with data from the triggering event.

### CODE REVIEW

**Comment by dfrommi** [comment]
Replaces a rather complex Hyperscript alternative, which required multiple events.
See https://github.com/bigskysoftware/htmx/discussions/1016 for more details

**Comment by jyrimatti** [comment]
Hi,

I'm currently using an extension described here (https://github.com/bigskysoftware/htmx/issues/1202) to replace query-path placeholders with data from htmx parameters.

These extensions solve a bit different problem, but they feel quite similar.
Any ideas if these could/should be somehow combined to be only a single extension?

**Comment by dfrommi** [comment]
If I understand it correctly, then the difference is the data-source (event vs params). 
I‘ve intentionally prefixed the variables in this PR with `event` to allow future enhancements without breaking existing usage. 
So my suggestion would be to just prefix params as well and replace it as a second step. 

Another difference is the usage of double vs single braces. I‘m open to both, just feel like single is a more common approach.

**Comment by jyrimatti** [comment]
Hi,

I just noticed Htmx already supports including event data as request parameters: https://htmx.org/attributes/hx-vals/
So I think all that is missing is an extension to "move" parameters to the path.
```html
 <div hx-ext="path-params" hx-get="/items/{id}" hx-vals='js:{id: event.detail.id}' hx-trigger="itemAdded from:body" />
```

I'm not a maintainer so I'm not saying what _should_ be done. But what do you think, would your problem be solved just with an extension like path-params? I could make a PR since I'm already using such an extension anyway.
(I don't really have an opinion about the amount of braces, you are probably right that single braces is more common in a URL template).

**Comment by jyrimatti** [comment]
Made a PR: https://github.com/bigskysoftware/htmx/pull/1307
Please let me know if you come up with any ideas for improvement!

**Comment by dfrommi** [comment]
Thanks @jyrimatti for the hint with the `hx-val` and also the PR.

Not sure if you've seen it in an earlier comment
> I‘ve intentionally prefixed the variables in this PR with event to allow future enhancements without breaking existing usage.
So my suggestion would be to just prefix params as well and replace it as a second step.

My approach would rather be to support `params` as a second data source for variables, rather than only params and copy everything to a param first. Just because it's one step less. So you would have something like `hx-get="/items/{params.id}"`

As I'm fairly new to HTMX, I'm very much looking forward to maintainer's opinion.

File: src/ext/path-vars.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+htmx.defineExtension('path-vars', {
+    onEvent: function (name, evt) {
+        if (name === "htmx:configRequest") {
+            let sourceEventData = evt.detail?.triggeringEvent?.detail;
```
[the remark this answers is not in the document]
Comment (dfrommi) [reply src/ext/path-vars.js]: Thanks for the feedback. I've updated the code using `? :`

**Comment by Telroshan** [comment]
Didn't notice before sorry, could you retarget your PR to the `dev` branch as per the [contribution guidelines](https://github.com/bigskysoftware/htmx/blob/master/CONTRIBUTING.md) ?
Also resolve that conflict with `extensions.md`

**Comment by dfrommi** [comment]
Well, I'd like to, but I'm struggling running tests at all.
Just to confirm, after `npm install`, I just need to run `npm test`, right? 

But then I always get 

```
> htmx.org@1.9.6 test
> mocha-chrome test/index.html

Promise Rejection:  Error: connect ECONNREFUSED ::1:53293
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1595:16) {
  errno: -61,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '::1',
  port: 53293
}
```

and that's where it remains stuck.

**Comment by alexpetros** [comment]
This is very similar to https://github.com/bigskysoftware/htmx/pull/1261, which I closed for the following reasons:

> While I understand the use-case, I'm not sure that dynamically (and silently) updating links is a good fit for the core competencies of htmx, which is primarily concerned with replacing elements based on server responses. The default htmx way to do this would be to just replace the entire container with a new DOM element, and new links.
>
> I'm sure there are htmx users that would want to extend htmx to do this kind of client-side munging, and that's a good fit for an extension, but probably not an officially supported one.

I do appreciate the effort put into this, and think it's a good extension. I recommend publishing it on your own GitHub, and we'll get a page up that links to some high-quality 3P extensions.

**Pull request closed without merging**